### PR TITLE
Pull request : Word cloud for the files containing word-frequency pair

### PR DESCRIPTION
**Pull request by satishsaley**
Hi Kenny,

I have added functionality to create word cloud if file already contains word-frequency pairs.

If the data from which we want to create a word cloud is large, we can run simple MapReduce job to calculate word-frequency pairs. Later, we will process the output of MapReduce job using kumo library to generate a word cloud.

Let me know how you find my changes. Please convey me any suggestions you have to make code more clean and robust.

### CODE REVIEW

**Comment by kennycason** [comment]
Thanks a lot, It looks like there are some conflicts with the FrequencyAnalyzer class due to some major recent changes I made :) 

You can try and rebase if you want, if not, I can as well! 

I have a few syntactic recommendations that I'll post in the PR.

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java [hunk 1]
```diff
@@ -48,6 +48,10 @@
     public List<WordFrequency> load(InputStream fileInputStream) throws IOException {
         return load(IOUtils.readLines(fileInputStream));
     }
+    
+    public List<WordFrequency> load(InputStream fileInputStream,FileType fileType) throws IOException {
```
Comment (kennycason) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: I like to add a space after the comma in the parameters.

InputStream fileInputStream, FileType fileType

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java [hunk 2]
```diff
@@ -95,7 +99,39 @@
         return sorted.subList(0, Math.min(sorted.size(), wordFrequencesToReturn));
     }
 
-    public void setStopWords(Collection<String> stopWords) {
+    public List<WordFrequency> load(final List<String> texts,FileType fileType) {
+    	if(fileType == FileType.REGULAR)
```
Comment (kennycason) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: I like to always wrap conditional statements in brackets. 
```java
if(fileType == FileType.REGULAR) {
    return load(texts);
}
```
for short-simple statements like this I also approve of:
```java
if(fileType == FileType.REGULAR) { return load(texts); }
```

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java [hunk 3]
```diff
@@ -95,7 +99,39 @@
...
+    		return load(texts);
+    	
+        final List<WordFrequency> wordFrequencies = new ArrayList<>();
+        // generate all word counts
+        final Map<String, Integer> cloud = calculateCloud(texts, wordTokenizer, fileType);
+
+        for(String key : cloud.keySet()) {
+            if(key.length() >= minWordLength
```
Comment (kennycason) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: you'll notice that I remove all the key.length checks and other normalization methods into another phase, after loading text. You'll notice this after a rebase. So just a forewarning.

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java [hunk 4]
```diff
@@ -95,7 +99,39 @@
...
+            }
+        }
+        return takeTopFrequencies(wordFrequencies);
+    }
+    
+    private Map<String, Integer> calculateCloud(List<String> texts, WordTokenizer tokenizer, FileType fileType) {
+    	final Map<String, Integer> cloud = new HashMap<>();
+        for(String text : texts) {
+            final List<String> words = Lambda.filter(stopWordFilter, tokenizer.tokenize(sanitizer.sanitize(text)));
+            int sz = words.size();
```
Comment (kennycason) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: i'd recommend a `final size = word.size()`. With the exception of incrementers in for-loops I rarely like to abbreviate variables :)

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java [hunk 5]
```diff
@@ -95,7 +99,39 @@
...
+            if(sz > 2)
```
Comment (kennycason) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: same comment with the brackets.

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java [hunk 6]
```diff
@@ -95,7 +99,39 @@
...
+            	continue;
+            for(int i=0; i<sz; i++){
```
Comment (kennycason) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: I typically prefer spacing as such:
```java
for(int i = 0; i < size; i++) { 
```

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java [hunk 7]
```diff
@@ -95,7 +99,39 @@
...
+        for(String key : cloud.keySet()) {
+            if(key.length() >= minWordLength
+                    && key.length() < MAX_LENGTH) {
+                wordFrequencies.add(new WordFrequency(key, cloud.get(key)));
+            }
+        }
+        return takeTopFrequencies(wordFrequencies);
+    }
+    
+    private Map<String, Integer> calculateCloud(List<String> texts, WordTokenizer tokenizer, FileType fileType) {
```
Comment (kennycason) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: Is FileType used in this method? I don't see it. If not, no need to pass it in.

**Comment by kennycason** [comment]
I have only one major issue that I think will need to be dealt with after rebasing. In your method for the KEYVALUE loading you still use the tokenizer. I *highly* recommend we don't do that. I think for the keyvalue format we should agree on a standard format. for example:
```
word: 10
dog: 1234
frog: 34454
```
Then we could split on the ":" character and trim the word/counts and build it form that. It will keep our implementation simple.

**Comment by kennycason** [comment]
Sorry for all the comments! I hope they aren't too much! Thanks for the contribution, and thanks for adding tests as well!

I overall do like the direction and think this will be useful for supporting json/yaml formats in the future as well. :)

**Comment by kennycason** [comment]
One more issue :)
I noticed after viewing the code in github that you appear to be using tab characters, currently I use spaces. Your editor should support replacing tabs with spaces (4).

**Comment by satishsaley** [comment]
Thank you Kenny for the comments. It would help me write cleaner code in future as well. I will make the changes and let you know.

**Comment by kennycason** [comment]
@satishsaley Awesome to hear! I didn't want to scare you away! :)

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java (hunk 7 above)
Comment (satishsaley) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: implemented

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java (hunk 6 above)
Comment (satishsaley) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: implemented

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java (hunk 5 above)
Comment (satishsaley) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: implemented

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java (hunk 4 above)
Comment (satishsaley) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: implemented

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java (hunk 2 above)
Comment (satishsaley) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: implemented

File: src/main/java/wordcloud/nlp/FrequencyAnalizer.java (hunk 1 above)
Comment (satishsaley) [inline src/main/java/wordcloud/nlp/FrequencyAnalizer.java]: implemented.

**Comment by satishsaley** [comment]
Hello Kenny,

I have made the changes you suggested. Let me know if I am missing something.

**Comment by kennycason** [comment]
Just wanted to let you know I haven't forgotten. I've been on a road trip :)

**Comment by satishsaley** [comment]
oh.cool. I will start road trip along CA-1 highway tomorrow. Where are you
travelling?

Thanks and Regards,
Satish Subhashrao Saley

**Comment by kennycason** [comment]
Arkansas -> Missouri -> Iowa -> Nebraska -> South Dakota [Mt. Rushmore] -> Montana -> Wyoming [Yellowstone (closed so i just camped in the rockies)] -> Idaho -> Washington (Seattle 1week) -> Oregon -> California [Redwood Forest, San Fransisco, Los Angeles (current location), Sequoia National Park] -> Arizona [Grand Canyon] -> New Mexico -> Texas [Dallas spa/korean town] -> Arkansas 

:) Quite the loop. I will then drive back to Seattle as I am moving there, but will go through Colorado/Nevada and just take a more central route in general.

CA-1 is a beautiful route!

**Pull request closed without merging**